pkg/cmd/server/origin/leaderelection.go

```diff
@@ -37,27 +36,34 @@ func NewLeaderElection(options configapi.MasterConfig, leader componentconfig.Le
 	id := fmt.Sprintf("master-%s", kutilrand.String(8))
 	name := "openshift-controller-manager"
 	namespace := "kube-system"
-	useEndpoints := false
+	useClientLock := false
+	var lock rl.Interface
 	if election := options.ControllerConfig.Election; election != nil {
-		if election.LockResource.Resource != "endpoints" || election.LockResource.Group != "" {
-			return nil, nil, fmt.Errorf("only the \"endpoints\" resource is supported for election")
-		}
 		name = election.LockName
 		namespace = election.LockNamespace
-		useEndpoints = true
-	}
+		useClientLock = true
 
-	lock := &rl.EndpointsLock{
-		EndpointsMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
-		Client:        kc.Core(),
-		LockConfig: rl.ResourceLockConfig{
-			Identity: id,
-		},
+		switch election.LockResource {
```

Comment:
Do you really need this check if `rl.New` will fail on it anyway?  Unless you are concerned that the group really must be `""`?